notebooks/02-ingestion/02-batch-ingestion.ipynb

```diff
@@ -0,0 +1,1008 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e0429f0e-ae98-472a-b686-8aa308985de6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "druid.sql.wait_until_ready('wikipedia_events')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ca8f811a-7795-4351-bc6b-3abb34da0116",
+   "metadata": {},
+   "source": [
+    "#### Query the Data\n",
+    "Let's take a look at the data that was loaded:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d2584e65-c952-47f1-a885-2e5d3a4fef2b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql = \"\"\"\n",
+    "SELECT channel, count(*) num_events\n",
+    "FROM \"wikipedia_events\" \n",
+    "GROUP BY 1 \n",
+    "ORDER BY 2 DESC \n",
+    "LIMIT 10\n",
+    "\"\"\"\n",
+    "druid.display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ae43d5f3-0350-40e6-b5a6-737c610d7562",
+   "metadata": {},
+   "source": [
+    "### Ingesting from Multiple Files"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7340d912-2441-48ef-b6ce-e182f57734d4",
+   "metadata": {},
+   "source": [
+    "[Druid Input Sources](https://druid.apache.org/docs/latest/ingestion/native-batch.html#splittable-input-sources) allow you to specify multiple files as input to an ingestion job.\n",
+    "\n",
+    "In the following example we are using the same file three times simulating multiple source files with the same schema. Normally this would be a list of different files to load: \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dd91d16b-1880-4a4c-8e8a-650075e26015",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql = '''\n",
+    "REPLACE INTO \"wikipedia_events_3\" OVERWRITE ALL\n",
+    "WITH \"ext\" AS \n",
+    "(\n",
+    "    SELECT *\n",
+    "    FROM TABLE(\n",
+    "      EXTERN(\n",
+    "        '{\"type\":\"http\",\n",
+    "          \"uris\":[ \"https://druid.apache.org/data/wikipedia.json.gz\",\n",
+    "                   \"https://druid.apache.org/data/wikipedia.json.gz\",\n",
+    "                   \"https://druid.apache.org/data/wikipedia.json.gz\"\n",
+    "                 ]\n",
+    "         }',\n",
+    "        '{\"type\":\"json\"}'\n",
+    "      )\n",
+    ") EXTEND (\"isRobot\" VARCHAR, \"channel\" VARCHAR, \"timestamp\" VARCHAR, \"flags\" VARCHAR, \"isUnpatrolled\" VARCHAR, \"page\" VARCHAR, \"diffUrl\" VARCHAR, \"added\" BIGINT, \"comment\" VARCHAR, \"commentLength\" BIGINT, \"isNew\" VARCHAR, \"isMinor\" VARCHAR, \"delta\" BIGINT, \"isAnonymous\" VARCHAR, \"user\" VARCHAR, \"deltaBucket\" BIGINT, \"deleted\" BIGINT, \"namespace\" VARCHAR, \"cityName\" VARCHAR, \"countryName\" VARCHAR, \"regionIsoCode\" VARCHAR, \"metroCode\" BIGINT, \"countryIsoCode\" VARCHAR, \"regionName\" VARCHAR))\n",
+    "SELECT\n",
+    "  TIME_PARSE(\"timestamp\") AS \"__time\",\n",
+    "  *\n",
```

Comment:
You could use this opportunity to flip to being specific about what columns are ingested.

Reply:
....moving on... trying to get this done, so I'm not doing this one.